umap/static/umap/js/umap.controls.js

```diff
@@ -741,21 +741,49 @@ L.U.Map.include({
   _openFacet: function () {
     const container = L.DomUtil.create('div', 'umap-facet-search'),
       title = L.DomUtil.add('h3', 'umap-filter-title', container, L._('Facet search')),
-      keys = Object.keys(this.getFacetKeys())
+      facetKeys = this.getFacetKeys(),
+      keys = Object.keys(facetKeys)
 
-    const knownValues = {}
+    const facetCriteria = {}
 
     keys.forEach((key) => {
-      knownValues[key] = []
-      if (!this.facets[key]) this.facets[key] = []
+      if (facetKeys[key]["type"] === "date") {
+        if (!facetCriteria[key]) facetCriteria[key] = {
+          "min": undefined,
+          "max": undefined
+        }
+        if (!this.facets[key]) this.facets[key] = {
+          "type": facetKeys[key]["type"],
+          "min": undefined,
+          "max": undefined
+        }
+      } else {
+        if (!facetCriteria[key]) facetCriteria[key] = {
+          "choices": []
+        }
+        if (!this.facets[key]) this.facets[key] = {
+          "type": facetKeys[key]["type"],
+          "choices": []
+        }
+      }
     })
 
     this.eachBrowsableDataLayer((datalayer) => {
       datalayer.eachFeature((feature) => {
         keys.forEach((key) => {
           let value = feature.properties[key]
-          if (typeof value !== 'undefined' && !knownValues[key].includes(value)) {
-            knownValues[key].push(value)
+          if (facetKeys[key]["type"] === "date") {
```

Comment:
Should we distinguish between `date` and `datetime` ? I'm thinking about the form UI, but maybe also the parsing process.

Reply:
Yes, again I agree. We should distinguish. But maybe go even further letting the user define the data type as well as the input field type. Since we have another conversation going on about this, I will resolve this conversation.

https://github.com/umap-project/umap/pull/1491#discussion_r1439898242